aptos-move/framework/src/natives/cryptography/groups/arkworks_backend.rs

```diff
@@ -0,0 +1,1896 @@
+            let point = ark_bls12_381::G2Affine::deserialize(bytes.as_slice());
+            match point {
+                Ok(point) => {
+                    let handle = context
+                        .extensions_mut()
+                        .get_mut::<Bls12381Context>()
+                        .add_g2_point(point.into_projective());
+                    Ok(NativeResult::ok(
+                        cost,
+                        smallvec![Value::bool(true), Value::u64(handle as u64)],
+                    ))
+                }
+                _ => {
+                    Ok(NativeResult::ok(
+                        cost,
+                        smallvec![Value::bool(false), Value::u64(0)],
+                    ))
+                }
+            }
+        }
+        "0x1::groups::BLS12_381_Gt" => {
+            let cost = (gas_params.bls12_381.fq12_deserialize + gas_params.bls12_381.fq12_pow_fr + gas_params.bls12_381.fq12_eq) * NumArgs::one();
+            let point = Fq12::deserialize(bytes.as_slice());
+            match point {
+                Ok(point) => {
+                    let r_buf = hex::decode(
+                        "01000000fffffffffe5bfeff02a4bd5305d8a10908d83933487d9d2953a7ed73",
+                    )
+                        .unwrap();
+                    let r =
+                        ark_ff::BigInteger256::deserialize_uncompressed(r_buf.as_slice()).unwrap();
+                    if Fq12::one() == point.pow(r) {
+                        let handle = context
+                            .extensions_mut()
+                            .get_mut::<Bls12381Context>()
+                            .add_gt_point(point);
+                        Ok(NativeResult::ok(
+                            cost,
+                            smallvec![Value::bool(true), Value::u64(handle as u64)],
+                        ))
+                    } else {
+                        Ok(NativeResult::ok(
+                            cost,
+                            smallvec![Value::bool(false), Value::u64(0)],
+                        ))
+                    }
+                }
+                _ => {
+                    Ok(NativeResult::ok(
+                        cost,
+                        smallvec![Value::bool(false), Value::u64(0)],
+                    ))
+                }
+            }
+        }
+        _ => {
+            Ok(NativeResult::err(InternalGas::zero(), E_UNKNOWN_GROUP))
+        }
+    }
+}
+
+fn deserialize_scalar_internal(
+    gas_params: &GasParameters,
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    assert_eq!(1, ty_args.len());
+    let type_tag = context
+        .type_to_type_tag(ty_args.get(0).unwrap())?
+        .to_string();
+    let bytes = pop_arg!(args, Vec<u8>);
+    match type_tag.as_str() {
+        "0x1::groups::BLS12_381_G1" | "0x1::groups::BLS12_381_G2" | "0x1::groups::BLS12_381_Gt" => {
```

Comment:
Are we gonna add an explicit `BLS12_381_Fr` type? 

Or just allow all operations that take a group element and a field element `s` of type `Scalar<T>` to work for any `T` in `"0x1::groups::BLS12_381_G1" | "0x1::groups::BLS12_381_G2" | "0x1::groups::BLS12_381_Gt"`?

Reply:
Yes we are going to have a `BLS12381_Fr`!